tests/test_fieldset.py

```diff
@@ -242,3 +247,23 @@ class MyParticle(ptype[mode]):
     elif dt_sign == -1:
         temp_theo = temp_vec[0]
     assert np.allclose(temp_theo, pset.particles[0].temp, atol=1e-5)
+
+
+def test_fieldset_defer_loading_with_diff_time_origin(tmpdir, filename='test_parcels_defer_loading'):
+    filepath = tmpdir.join(filename)
+    data0, dims0 = generate_fieldset(10, 10, 1, 10)
+    dims0['time'] = np.arange(0, 10, 1) * 3600
+    fieldset_out = FieldSet.from_data(data0, dims0)
+    fieldset_out.U.grid.time_origin = np.datetime64('2018-04-20')
+    data1, dims1 = generate_fieldset(10, 10, 1, 10)
+    dims1['time'] = np.arange(0, 10, 1) * 1800 + (24+25)*3600
+    gridW = RectilinearZGrid(dims1['lon'], dims1['lat'], dims1['depth'], dims1['time'])
+    fieldW = Field('W', np.zeros(data1['U'].shape), grid=gridW)
+    fieldW.grid.time_origin = np.datetime64('2018-04-18')
+    fieldset_out.add_field(fieldW)
+    fieldset_out.write(filepath)
+    fieldset = FieldSet.from_parcels(filepath, extra_fields={'W': 'W'})
+    pset = ParticleSet.from_list(fieldset, pclass=JITParticle, lon=[0.5], lat=[0.5], depth=[0.5],
+                                 # time=[np.datetime64('2018-04-20 01:00')])
```

Comment:
I would like to start using np.datetime64 in dates. This comment is waiting for next branch which is coming. It will be uncommented and replace the next line.